x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/NumericAggregate.java

```diff
@@ -40,15 +40,15 @@ protected TypeResolution resolveType() {
                 sourceText(),
                 DEFAULT,
                 "datetime",
-                "numeric except unsigned_long"
+                "numeric except unsigned_long or counter_long or counter_integer or counter_double"
```

Comment:
Nit: `except [unsigned_long, counter_*(double, integer, long)]`

Compacted the exception message a bit.

Reply:
++, I pushed https://github.com/elastic/elasticsearch/pull/107877/commits/165385ef238809644ffcb92228afce0ea1bfdaa4